src/pages/iou/steps/IOUParticipantsPage/IOUParticipantsRequest.js

```diff
@@ -94,7 +94,7 @@ class IOUParticipantsRequest extends Component {
             indexOffset: 0,
         });
 
-        if (this.state.userToInvite && !isCurrentUser(this.state.userToInvite)) {
+        if (this.state.userToInvite && !isExcludedIOUUsers(this.state.userToInvite)) {
```

Comment:
This looked good to me initially, but I am now second guessing it. It feels like it would be more appropriate to just not return a `userToInvite` from `getOptions()`. 

Then we would not need to import this method into these two places and would probably not need the method at all. Just don't allow setting `userToInvite` if they are one of the emails:

https://github.com/Expensify/App/blob/95bf36d950f57d01ae37af2e0d8f2e664a273b75/src/libs/OptionsListUtils.js#L505-L525

We'd still need to do some kind of boolean flag like `shouldInviteExpensifyEmails` passed to `getOptions()` as there are some cases where we do want to be able to invite these users.

Reply:
There are already separate flags, we could reuse them or add a separate flag?
https://github.com/Expensify/App/blob/95bf36d950f57d01ae37af2e0d8f2e664a273b75/src/libs/OptionsListUtils.js#L325-L327

1. If we are going to reuse, then modify the line 
https://github.com/Expensify/App/blob/95bf36d950f57d01ae37af2e0d8f2e664a273b75/src/libs/OptionsListUtils.js#L512
as follows,
```
 && (!_.find(loginOptionsToExclude, loginOptionToExclude => loginOptionToExclude.login.toLowerCase() === searchValue.toLowerCase())) 
```

2. If we are going to add a separate boolean `shouldInviteExpensifyEmails` with default value as `false`
```
&& (!shouldInviteExpensifyEmails && !_.contains(CONST.EXPENSIFY_EMAILS, searchValue.toLowerCase()))
```
Either way, we can just remove the `isExcludedIOUUsers` method.

What can we do here?
@marcaaron post your thoughts, I'll update PR, once you are okay with the proposed changes!

Reply:
@marcaaron 
> I'm confused now. So there is already logic to exclude them but why are they are not getting excluded? Why do they show in the list then? 😦
Because we do only `===` not check without about case sensitivity 

Check with the current staging app, search `receipts@expensify.com` it won't show upresults. But search `Receipts@expensify.com` it will show up

Reply:
Hmm I checked on `main` branch. The bug we are "fixing" seems already fixed. Putting this PR on hold.

We can fix the casing issue, but that's not what we set out to do (and is a minor detail of the larger issue presented). So let's discuss how to proceed on the issue? Thanks!